### PR TITLE
Resolved issue with empty get_xxx calls

### PR DESCRIPTION
**Pull request by PeterMeisrimelModelon**

### CODE REVIEW

File: src/pyfmi/fmi.pyx [hunk 1]
```diff
@@ -1589,6 +1589,8 @@ cdef class FMUModelBase(ModelBase):
         nref = len(val_ref)
         cdef N.ndarray[FMIL.fmi1_real_t, ndim=1,mode='c'] val = N.array([0.0]*nref, dtype=float, ndmin=1)
 
+        if not nref: ## get_real([])
```
Comment (chria) [inline src/pyfmi/fmi.pyx:1592]: As nref is of type "size_t" I think it is more clear to directly compare against 0. i.e:

```
if nref == 0:
   ...
```
It it was a general object, then I would agree with this syntax.

File: src/pyfmi/fmi.pyx [hunk 2]
```diff
@@ -1805,17 +1814,21 @@ cdef class FMUModelBase(ModelBase):
 
         Calls the low-level FMI function: fmiGetString
         """
-        cdef int         status
+        cdef int status
         cdef N.ndarray[FMIL.fmi1_value_reference_t, ndim=1, mode='c'] input_valueref = N.array(valueref, dtype=N.uint32, ndmin=1).ravel()
-        cdef FMIL.fmi1_string_t* output_value = <FMIL.fmi1_string_t*>FMIL.malloc(sizeof(FMIL.fmi1_string_t)*input_valueref.size)
+        cdef int nref = input_valueref.size
```
Comment (chria) [inline src/pyfmi/fmi.pyx:1819]: nref should be of type size_t:
`cdef FMIL.size_t nref`

File: src/pyfmi/fmi.pyx [hunk 3]
```diff
@@ -4199,12 +4212,15 @@ cdef class FMUModelBase2(ModelBase):
 
         Calls the low-level FMI function: fmi2GetReal
         """
-        cdef int         status
-
+        cdef int status
         cdef N.ndarray[FMIL.fmi2_value_reference_t, ndim=1,mode='c'] input_valueref = N.array(valueref, copy=False, dtype=N.uint32).ravel()
-        cdef N.ndarray[FMIL.fmi2_real_t, ndim=1,mode='c']            output_value   = N.zeros(input_valueref.size)
+        cdef int nref = input_valueref.size
```
Comment (chria) [inline src/pyfmi/fmi.pyx:4217]: `cdef FMIL.size_t nref`

File: src/pyfmi/fmi.pyx [hunk 4]
```diff
@@ -4441,17 +4462,22 @@ cdef class FMUModelBase2(ModelBase):
 
         Calls the low-level FMI function: fmi2GetString
         """
-        cdef int         status
+        cdef int status
         cdef N.ndarray[FMIL.fmi2_value_reference_t, ndim=1, mode='c'] input_valueref = N.array(valueref, dtype=N.uint32, ndmin=1).ravel()
-        cdef FMIL.fmi2_string_t* output_value = <FMIL.fmi2_string_t*>FMIL.malloc(sizeof(FMIL.fmi2_string_t)*input_valueref.size)
+        cdef int nref = input_valueref.size
```
Comment (chria) [inline src/pyfmi/fmi.pyx:4467]: `cdef FMIL.size_t nref`

**Review by chria — COMMENTED** [review]
Looks good! Just minor comments.

File: src/pyfmi/fmi.pyx (hunk 1 above)
Comment (PeterMeisrimelModelon) [reply src/pyfmi/fmi.pyx]: Fixed.

File: src/pyfmi/fmi.pyx (hunk 2 above)
Comment (PeterMeisrimelModelon) [reply src/pyfmi/fmi.pyx]: Fixed, also the others.

**Comment by chria** [comment]
Looks good!

**Pull request merged**